### PR TITLE
fix: add username protection

### PR DESCRIPTION
**Pull request by rebelchris**
### Describe what this PR does
- Ensure protection from FE side to validate username

Related: https://github.com/dailydotdev/daily-api/pull/1776

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-{number} #done

### Preview domain
https://hotfix-username-protection.preview.app.daily.dev

### CODE REVIEW

File: packages/shared/src/hooks/useProfileForm.ts [hunk 1]
```diff
@@ -56,6 +56,8 @@ export const onValidateHandles = (
     if (!isValid) {
       return { username: errorMessage.profile.invalidUsername };
     }
+  } else {
+    return { username: errorMessage.profile.invalidUsername };
   }
```
[the remark this answers is not in the document]
Comment (omBratteng) [reply packages/shared/src/hooks/useProfileForm.ts:61]: https://hotfix-username-protection.preview.app.daily.dev/account/profile

[the remark this answers is not in the document]
Comment (omBratteng) [reply packages/shared/src/hooks/useProfileForm.ts:61]:
```suggestion
  } else if (!after.username) {
    return { username: errorMessage.profile.invalidUsername };
  }
```

[the remark this answers is not in the document]
Comment (rebelchris) [reply packages/shared/src/hooks/useProfileForm.ts]: Ah yes missed it in my testing somehow

**Review by omBratteng — APPROVED** [review]
Tested on preview, and it now prevents me from setting no username, and I can still change it.

**Review by nensidosari — APPROVED** [review]

**Review by omBratteng — APPROVED** [review]

**Comment by rebelchris** [comment]
@omBratteng yeah forgot we use it as a generic update 😅 (bit silly design maybe)
So now check if the key exists and only then force check :)

**Pull request merged**